### PR TITLE
feat: add node graceful shutdown kubelet config

### PR DESCRIPTION
**Pull request by weichch**
**What type of PR is this?**

/kind feature
/kind api-change

**What this PR does / why we need it**:

This PR adds node graceful shutdown Kubelet config items and exposes those for client use, so that we can allow customers to enable this feature on their nodes.

This PR is recreated from the previous PR (#4147) from fork as per comment.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by weichch** [comment]
As per mentioned in #4147, the feature gate is [by default enabled](https://kubernetes.io/blog/2021/04/21/graceful-node-shutdown-beta/#how-do-i-use-it) in all k8s versions we currently support.

**Review by cameronmeissner — APPROVED** [review]
approved pending agentbaker e2es

**Review by cameronmeissner — APPROVED** [review]

File: pkg/agent/datamodel/types.go
Comment (UtheMan) [inline pkg/agent/datamodel/types.go:2053]: Where do we validate whether `ShutdownGracePeriodCriticalPods` is lower than `ShutdownGracePeriod`?

Comment (weichch) [reply pkg/agent/datamodel/types.go:2053]: Validation is going to be in RP, similar to `ImageGcHighThreshold` and `ImageGcLowThreshold`.

File: pkg/agent/utils.go
Comment (weichch) [inline pkg/agent/utils.go:500]: I also wanted to add some defaulting logic based on Kubernetes version. What do you think if we add `IsKubernetesVersionGe(xxx)` in this method? @UtheMan

**Comment by weichch** [comment]
I'm putting this on hold now as there might be some investigations I need to do *before* this can be merged.

**Comment by cameronmeissner** [comment]
@weichch closing for now due to inactivity, feel free to reopen when needed

**Pull request closed without merging**
